### PR TITLE
Change `char`->`free` to `auto_char char` for autocleanup

### PR DESCRIPTION
**Pull request by H3rnand3zzz**
See commit messages. First part of big cleanup (#1858 ).

# I ran valgrind when using my new feature
No

- [x] Manual review
- [ ] Run Valgrind

### CODE REVIEW

**Comment by jubalh** [comment]
Please adjust commit message accordingly. I think this description is much more descriptive of what you are trying to accomplish `Move from `char` to `auto_char` to automatically free it`.

**Review by jubalh — CHANGES_REQUESTED** [review]
Now you started to split the commits into several PRs, which is okay.
But inside the commit you still do several unrelated things.
Please change that and see https://github.com/profanity-im/profanity/pull/1858#issuecomment-1621373770

**Comment by H3rnand3zzz** [comment]
Maybe I will do it later, right now I am unable to.

**Comment by jubalh** [comment]
> Maybe I will do it later, right now I am unable to.

I made the changes as discussed via PM.

**Review by jubalh — CHANGES_REQUESTED** [review]
you remove _is_correct_plugin_extension etc. probably mistake during your rebase on master

File: src/command/cmd_funcs.c [hunk 1]
```diff
@@ -188,44 +188,41 @@ _string_matches_one_of(const char* what, const char* is, bool is_can_be_null, co
...
 
-/*
- * Take a line of input and process it, return TRUE if profanity is to
- * continue, FALSE otherwise
+/**
+ * @brief Processes a line of input and determines if profanity should continue.
+ *
+ * @param window
+ * @param inp The input string to process.
+ * @return Returns TRUE if profanity is to continue, FALSE otherwise.
```
Comment (jubalh) [inline src/command/cmd_funcs.c:196]: I don't see what advantage this comment has over the one we already had.
It's just more lengthy.

File: src/command/cmd_funcs.c [hunk 2]
```diff
@@ -9594,30 +9533,12 @@ _url_aesgcm_method(ProfWin* window, const char* cmd_template, gchar* url, gchar*
 }
 #endif
 
-static gboolean
```
Comment (jubalh) [inline src/command/cmd_funcs.c:9597]: Why is `_download_install_plugin()` deleted?

File: src/xmpp/connection.c [hunk 3]
```diff
@@ -1108,11 +1103,10 @@ _xmpp_file_logger(void* const userdata, const xmpp_log_level_t xmpp_level, const
 static void
 _random_bytes_init(void)
 {
-    char* rndbytes_loc;
+    auto_gchar gchar* rndbytes_loc = files_get_data_path(FILE_PROFANITY_IDENTIFIER);
+    ;
```
Comment (jubalh) [inline src/xmpp/connection.c:1107]: `;`

**Review by jubalh — CHANGES_REQUESTED** [review]
Please see comments.
And osx CI:
```
src/command/cmd_funcs.c:135:1: error: function '_download_install_plugin' has internal linkage but is not defined [-Werror,-Wundefined-internal]
_download_install_plugin(ProfWin* window, gchar* url, gchar* path);
^
src/command/cmd_funcs.c:7055:9: note: used here
        _download_install_plugin(window, args[1], NULL);
        ^
```

File: src/command/cmd_funcs.c (hunk 2 above)
Comment (jubalh) [reply src/command/cmd_funcs.c]: @H3rnand3zzz I edited your commit and added the function back

File: src/xmpp/connection.c (hunk 3 above)
Comment (jubalh) [reply src/xmpp/connection.c]: @H3rnand3zzz I removed that now

**Comment by jubalh** [comment]
@H3rnand3zzz can you please review this PR? I 2 above mentioned edits.

**Comment by H3rnand3zzz** [comment]
> @H3rnand3zzz can you please review this PR? I 2 above mentioned edits.

those are correct changes, thanks

File: src/command/cmd_funcs.c (hunk 1 above)
Comment (jubalh) [reply src/command/cmd_funcs.c:196]: Re doxygen please see https://github.com/profanity-im/profanity/issues/1859

File: src/database.c [hunk 4]
```diff
@@ -126,7 +125,6 @@ log_database_init(ProfAccount* account)
     }
 
     log_debug("Initialized SQLite database: %s", filename);
-    free(filename);
     return TRUE;
 
 out:
```
[the remark this answers is not in the document]
Comment (jubalh) [reply src/database.c:130]: Would be good to test this before we can merge

File: src/otr/otr.c [hunk 5]
```diff
@@ -319,8 +315,7 @@ otr_on_message_recv(const char* const barejid, const char* const resource, const
     if (policy == PROF_OTRPOLICY_ALWAYS && *decrypted == FALSE && !whitespace_base) {
         char* otr_query_message = otr_start_query();
         cons_show("Attempting to start OTR session…");
-        char* id = message_send_chat_otr(barejid, otr_query_message, FALSE, NULL);
-        free(id);
+        free(message_send_chat_otr(barejid, otr_query_message, FALSE, NULL));
```
[the remark this answers is not in the document]
Comment (jubalh) [reply src/otr/otr.c:318]: lets leave it for now to get on with this

**Review by jubalh — APPROVED** [review]
Should be ok now

**Comment by jubalh** [comment]
Thanks for your effort!

**Pull request merged**